src/ape/managers/compilers.py

```diff
@@ -42,6 +43,21 @@ def __getattr__(self, name: str) -> Any:
 
         raise ApeAttributeError(f"No attribute or compiler named '{name}'.")
 
+    @property
+    def compilers_json_file(self) -> Path:
+        return self.project_manager.local_project._cache_folder / "compilers.json"
+
+    @property
+    def compilers_json(self) -> Dict[str, Compiler]:
```

Comment:
- [ ] needed version layer

Reply:
so the keys here are arbitrary, can be something like `{compiler plugin}-{compiler version}` so that if more than one version is required then the key is supported